### PR TITLE
Fix bot sounds being inaccurate, fix TP bots stuttering after teleporting, fix bot wrongfully spawning, fix gokz-quiet not working on bots sometimes

### LINKED ISSUE #403: Jumpstat replays spawn a bot and doesn't remove it on a failed loading
When trying to load any jumpstat replay and it fails to do (wrong map/no replay), the bot is gonna be spawned and will remain alive until the map change/server restart. Latest GOKZ version as of the current date.

### PR DESCRIPTION
**Pull request by zer0k-z**
For the first issue, we SDKCall the jumping sound function whenever the bot "jumps", and sometimes the bot's movetype is incorrect.

The second issue is resolved by setting the absolute origins and velocity of the bots because `TeleportEntity` does not do it.

The third issue is resolved by making sure the replay exists before spawning the bot. (Fixes #403)

The last issue ensures gokz-quiet works when bot spawns.

### CODE REVIEW

File: addons/sourcemod/scripting/gokz-replays/playback.sp [hunk 1]
```diff
@@ -937,12 +938,15 @@ void PlaybackVersion2(int client, int bot, int &buttons)
...
-		Movement_GetOrigin(client, currentOrigin);
-		MakeVectorFromPoints(currentOrigin, currentTickData.origin, velocity);
-		ScaleVector(velocity, 1.0 / GetTickInterval());
-		TeleportEntity(client, NULL_VECTOR, currentTickData.angles, velocity);
+		TeleportEntity(client, NULL_VECTOR, currentTickData.angles, currentTickData.velocity);
+		// TeleportEntity does not set the absolute origin and velocity so we need to do it
+		// to prevent inaccurate eye position interpolation.
+		SetEntPropVector(client, Prop_Data, "m_vecAbsVelocity", currentTickData.velocity);
+		
+		SetEntPropVector(client, Prop_Data, "m_vecAbsOrigin", currentTickData.origin);
```
Comment (zealain) [inline addons/sourcemod/scripting/gokz-replays/playback.sp:946]: Setting the origin directly instead of changing only the velocity was usually the cause of stuttering in the past, would it be enough to fix the issue if we only set the abs velocities and eye angles?

**Review by zealain — CHANGES_REQUESTED** [review]

File: addons/sourcemod/scripting/gokz-replays/playback.sp [hunk 2]
```diff
@@ -937,12 +938,15 @@ void PlaybackVersion2(int client, int bot, int &buttons)
 			EmitSoundToClientSpectators(client, gC_ModeEndSounds[GOKZ_GetCoreOption(client, Option_Mode)]);
 		}
 
-		// Set velocity to travel from current origin to recorded origin
-		float currentOrigin[3], velocity[3];
-		Movement_GetOrigin(client, currentOrigin);
-		MakeVectorFromPoints(currentOrigin, currentTickData.origin, velocity);
-		ScaleVector(velocity, 1.0 / GetTickInterval());
-		TeleportEntity(client, NULL_VECTOR, currentTickData.angles, velocity);
+		TeleportEntity(client, NULL_VECTOR, currentTickData.angles, currentTickData.velocity);
```
[the remark this answers is not in the document]
Comment (zer0k-z) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:941]: Calculating it from the origin will fail if the player gets teleported as the velocity becomes inaccurate, causing weird interpolation issues and/or sound issues

I was unaware of this issue until now, what happened previously?

File: addons/sourcemod/scripting/gokz-replays/playback.sp (hunk 1 above)
Comment (zer0k-z) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:946]: The current stuttering in 3.3.0 is caused by client interpolation not working on fake clients that aren't bots. GameChaos's SimpleTAS also does a similar thing so I believe it should be sufficient.

For the record, the [kz_kiwiterror run on youtube](https://www.youtube.com/watch?v=fa8fZ8J56hg) is recorded with the code in this PR. Of course you should also test it for yourself as well, as I might miss some other edge cases.

File: addons/sourcemod/scripting/gokz-replays/playback.sp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zealain) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:941]: If I remember correctly, the bot would start to drift off in some cases, I assume due to rounding differences, but it's been a hot minute since we last tried that. Would the interpolation issues be solved if we only took the replay velocity on teleport ticks? We can directly set the origin as well in those cases if that'd help I would say.

File: addons/sourcemod/scripting/gokz-replays/playback.sp (hunk 1 above)
Comment (zealain) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:946]: I don't know how `m_vecAbsOrigin` really behaves, I just know that setting the regular origin every tick is no good, so it raised a red flag. If it is required to fix the stutters we'll leave it, I'll do some testing as well.

Comment (zer0k-z) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:946]: I think as long as the game doesn't add the "no interpolation" flag to the bot (might be the case with `TeleportEntity` using the `origin` argument), it should be fine, hopefully.

**Comment by Ciallo-Ani** [comment]
could you consider using an easier way to resolve it?
like https://github.com/shavitush/bhoptimer/blob/master/addons/sourcemod/scripting/shavit-replay-playback.sp#L2152-L2175

File: addons/sourcemod/scripting/gokz-replays/playback.sp (hunk 1 above)
Comment (Ciallo-Ani) [inline addons/sourcemod/scripting/gokz-replays/playback.sp:946]: damn you should not set their origin by every ticks, or your bot will be choppy whatever it's fakeclient or not.
here's a simple solution: https://github.com/shavitush/bhoptimer/blob/master/addons/sourcemod/scripting/shavit-replay-playback.sp#L2677-L2689

Comment (Ciallo-Ani) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:946]: damn you should not set their origin by every ticks, or your bot will be choppy whatever it's fakeclient or not.
here's a simple solution: https://github.com/shavitush/bhoptimer/blob/master/addons/sourcemod/scripting/shavit-replay-playback.sp#L2677-L2689

**Comment by zer0k-z** [comment]
The landing and jumping issue that this PR is referring to is different from that fix I believe. His fix refers to the issue that the bot makes no footstep sounds caused by them being in noclip most of the time.

In GOKZ, bots are rarely ever in noclip, and they do make footsteps and ladder sounds correctly. 

The first issue is that the bot never actually "jump" and therefore never makes any jumping sound. This wasn't very noticeable until I recorded kiwiterror and didn't hear the jumping sounds in the low ceiling bhop sections which annoyed me.

The second issue is the bot plays the "falling from very high" landing sound while they obviously did not in reality. #130 is the issue, it's closed but it actually still happens and I only managed to find the solution recently.

File: addons/sourcemod/scripting/gokz-replays/playback.sp (hunk 1 above)
Comment (zer0k-z) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:946]: I doubt that changing one netprop would cause a lot of lag, and it's definitely less expensive than TeleportEntity with an origin argument (and it only changes the local origin which is insufficient as well, as described in the PR)

I could in theory check for the predicted origin vs the actual replay origin and teleport them when necessary but then the extra comparisons would cost more resources either way.

File: addons/sourcemod/scripting/gokz-replays/playback.sp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zer0k-z) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:941]: As the origin of the bot gets updated every tick, drifting should not happen.
About taking velocity during teleport ticks only, this might cause issues if you get teleported by the maps themselves by a short distance as those aren't recorded into the replays themselves, and as far as the client is concerned, being teleported by a map or by the menu should be the same to them.

[the remark this answers is not in the document]
Comment (Ciallo-Ani) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:941]: teleported means bot's velocity becoms very high, why not just teleport them at that tick?

File: addons/sourcemod/scripting/gokz-replays/playback.sp (hunk 1 above)
Comment (Ciallo-Ani) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:946]: I noticed that the bot is a bit choppy after setting origin(netprop) on every tick. Haven't you noticed?

**Comment by zer0k-z** [comment]
I have pushed a new commit that also changes the non absolute velocity and origin of the bot, which turns out to be necessary to prevent choppiness during the playback and I forgot to put them into the pull request.

For more information, [I quickly recorded the bot at 0.2 timescale](https://youtu.be/xjtzLqzrG5Y) to ensure that it is as smooth as possible.

The whole thing on the left is `cl_pdump` which I used to track the networked values to the client. It is useful to see how a replay bot differs from a real player in terms of netprops.

The console on the right is running `cl_interp_watch`, which is responsible for tracking client interpolation. In 3.3.0, tracking the bot would not print out anything, as it just straight up does not interpolate non-bot fake clients.

From the footage, setting the origins and velocity do not prevent the interpolation from working, and the bot does seem smooth as intended. The only non smooth part of the video is caused by either teleport, or ducking which immediately raises/lowers the player origin by 9.

On a side note, this PR might also fixes a small issue where the bot's view would clip into a ceiling for one tick if it ducks next to it, but I'm not sure about that.

File: addons/sourcemod/scripting/gokz-replays/playback.sp (hunk 1 above)
Comment (zer0k-z) [reply addons/sourcemod/scripting/gokz-replays/playback.sp:947]: The latest commit should make the bot much less choppy now.

**Comment by zer0k-z** [comment]
I pushed a new commit that fixes issue where adding bots wouldn't work on servers with too many players on one certain team. This also fixes a bug where the timer sound is not played if the player paused before they press the end timer.

**Comment by zealain** [comment]
@zer0k-z Would it be possible to seperate the bot TP part from the rest of the PR? I still don't know why setting the origin every tick would now magically work when it never did in the past and want to do more thorough testing on that first, but I'm not sure when I would have time to do that properly. If the PR would be split, I could at least merge the rest (which was requested by someone to prioritize).

**Comment by zer0k-z** [comment]
Deprecated and replaced by the mentioned PRs.
